go/libraries/doltcore/env/actions/table.go

```diff
@@ -21,74 +21,127 @@ import (
 	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
 )
 
-func CheckoutAllTables(ctx context.Context, dEnv *env.DoltEnv) error {
+func CheckoutAllTables(ctx context.Context, dEnv *env.DoltEnv) (unknown string, err error) {
 	roots, err := getRoots(ctx, dEnv, WorkingRoot, StagedRoot, HeadRoot)
 
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	tbls, err := AllTables(ctx, roots[WorkingRoot], roots[StagedRoot], roots[HeadRoot])
 
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return checkoutTables(ctx, dEnv, roots, tbls)
+	docs := *env.AllValidDocDetails
+
+	return checkoutTablesAndDocs(ctx, dEnv, roots, tbls, docs)
 
 }
 
-func CheckoutTables(ctx context.Context, dEnv *env.DoltEnv, tbls []string) error {
+func CheckoutTablesAndDocs(ctx context.Context, dEnv *env.DoltEnv, tbls []string, docs []doltdb.DocDetails) (unknown string, err error) {
 	roots, err := getRoots(ctx, dEnv, WorkingRoot, StagedRoot, HeadRoot)
 
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return checkoutTables(ctx, dEnv, roots, tbls)
+	return checkoutTablesAndDocs(ctx, dEnv, roots, tbls, docs)
 }
 
-func checkoutTables(ctx context.Context, dEnv *env.DoltEnv, roots map[RootType]*doltdb.RootValue, tbls []string) error {
-	var unknown []string
+func checkoutTablesAndDocs(ctx context.Context, dEnv *env.DoltEnv, roots map[RootType]*doltdb.RootValue, tbls []string, docs []doltdb.DocDetails) (unknown string, err error) {
+	unknownTbls := []string{}
 
 	currRoot := roots[WorkingRoot]
 	staged := roots[StagedRoot]
 	head := roots[HeadRoot]
+
+	if len(docs) > 0 {
+		currRoot, staged, err = getUpdatedWorkingAndStagedWithDocs(ctx, dEnv, currRoot, staged, head, docs)
+		if err != nil {
+			return "", err
+		}
+	}
+
 	for _, tblName := range tbls {
+		if tblName == doltdb.DocTableName {
+			continue
+		}
 		tbl, ok, err := staged.GetTable(ctx, tblName)
 
 		if err != nil {
-			return err
+			return "", err
 		}
 
 		if !ok {
 			tbl, ok, err = head.GetTable(ctx, tblName)
 
 			if err != nil {
-				return err
+				return "", err
 			}
 
 			if !ok {
-				unknown = append(unknown, tblName)
+				unknownTbls = append(unknownTbls, tblName)
 				continue
 			}
 		}
 
 		currRoot, err = currRoot.PutTable(ctx, tblName, tbl)
 
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
-	if len(unknown) > 0 {
+	if len(unknownTbls) > 0 {
+		for _, tbl := range unknownTbls {
+			if has, err := currRoot.HasTable(ctx, tbl); err != nil {
+				return "", err
+			} else if !has {
+				return tbl, doltdb.ErrTableNotFound
```

Comment:
This function is now returning two error params, essentially. Rather than do that, you should just return a descriptive enough error here, one that includes the name of the table not found.

Reply:
Looks like actions.NewTblNotExistError is what you want

Reply:
Using NewTblNotExistError! https://github.com/liquidata-inc/dolt/pull/299/files#diff-7bbaaa4ef160f9fffed44cef87371fa5R120

Just returning one error param now.